tests/func/test_data_cloud.py

```diff
@@ -308,6 +310,73 @@ def test_pull_git_imports(tmp_dir, dvc, scm, erepo):
     assert (tmp_dir / "new_dir" / "bar").read_text() == "bar"
 
 
+@pytest.mark.parametrize(
+    "erepo", [pytest.lazy_fixture("git_dir"), pytest.lazy_fixture("erepo_dir")]
+)
+def test_pull_git_imports_with_store(tmp_dir, dvc, scm, erepo, local_remote):
+    with erepo.chdir():
+        erepo.scm_gen({"dir": {"bar": "bar"}}, commit="second")
+        erepo.scm_gen("foo", "foo", commit="first")
+
+    dvc.imp(os.fspath(erepo), "foo", store=True)
+    dvc.imp(os.fspath(erepo), "dir", out="new_dir", store=True, rev="HEAD~")
+
+    assert dvc.pull()["fetched"] == 0
+    assert dvc.push() == 3
+
+    for item in ["foo", "new_dir", dvc.cache.local.cache_dir]:
+        remove(item)
+    os.makedirs(dvc.cache.local.cache_dir, exist_ok=True)
+    clean_repos()
+
+    # Close the repo since otherwise Windows holds references open which
+    # prevent deletion
+    erepo.close()
+    remove(erepo)
+
+    assert dvc.pull(force=True)["fetched"] == 2
+
+    assert (tmp_dir / "foo").exists()
+    assert (tmp_dir / "foo").read_text() == "foo"
+
+    assert (tmp_dir / "new_dir").exists()
+    assert (tmp_dir / "new_dir" / "bar").read_text() == "bar"
+
+
+def test_store_false_pull_error(tmp_dir, dvc, scm, local_remote, caplog):
```

Comment:
Hmm, I wonder whether we need any communication with user (the "out" is not stored) at all. Do we actually have a situation, where this information would pop up? 

If we `import` without `--store` option, DVC (upon pull) will try to re-import the file. So there will actually be no situation when we try to checkout `out` from cache, that, at the same time is not stored.

Reply:
Sure, unless you've e.g. run `git clean -xdf` and wiped your cache, or cloned someone else's repository. I hope the situation wouldn't happen often, but it is possible so we ought to have a test with some kind of well-defined behavior for that scenario, whatever that behavior is.